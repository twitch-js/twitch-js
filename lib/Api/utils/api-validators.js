"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.apiOptions = void 0;
var invariant_1 = __importDefault(require("invariant"));
var conformsTo_1 = __importDefault(require("lodash/conformsTo"));
var defaults_1 = __importDefault(require("lodash/defaults"));
var isFunction_1 = __importDefault(require("lodash/isFunction"));
var isString_1 = __importDefault(require("lodash/isString"));
var isUndefined_1 = __importDefault(require("lodash/isUndefined"));
exports.apiOptions = function (options) {
    var shape = {
        token: isString_1.default,
        clientId: isString_1.default,
        onAuthenticationFailure: function (cb) { return isFunction_1.default(cb) || isUndefined_1.default(cb); },
    };
    options = defaults_1.default(options, {
        clientId: undefined,
        onAuthenticationFailure: undefined,
    });
    invariant_1.default(conformsTo_1.default(options, shape), 'Expected valid options');
    return options;
};
//# sourceMappingURL=api-validators.js.map