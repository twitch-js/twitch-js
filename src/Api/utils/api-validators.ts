import invariant from 'invariant'

import conformsTo from 'lodash/conformsTo'
import defaults from 'lodash/defaults'
import isFunction from 'lodash/isFunction'
import isString from 'lodash/isString'
import isUndefined from 'lodash/isUndefined'

import { ApiOptions } from '../api-types'

export const apiOptions = (options: Partial<ApiOptions>): ApiOptions => {
  const shape = {
    token: isString,
    clientId: isString,
    onAuthenticationFailure: (cb: unknown) => isFunction(cb) || isUndefined(cb),
  }

  options = defaults(options, {
    clientId: undefined,
    onAuthenticationFailure: undefined,
  })

  invariant(conformsTo(options, shape), 'Expected valid options')

  return options as ApiOptions
}
