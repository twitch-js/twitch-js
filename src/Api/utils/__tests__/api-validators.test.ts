import * as validators from '../api-validators'

describe('Api/utils/validators', () => {
  const options = { token: 'TOKEN', clientId: 'CLIENT_ID' }

  describe('apiOptions', () => {
    test('should return default chat options', () => {
      const actual = validators.apiOptions(options)

      expect(actual).toMatchSnapshot()
    })

    test('should throw if token is missing', async () => {
      expect(() =>
        validators.apiOptions({ ...options, token: undefined }),
      ).toThrow()
    })

    test('should throw if clientId is missing', async () => {
      expect(() =>
        validators.apiOptions({ ...options, clientId: undefined }),
      ).toThrow()
    })
  })
})
