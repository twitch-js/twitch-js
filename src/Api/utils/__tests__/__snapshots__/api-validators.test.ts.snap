// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api/utils/validators apiOptions should return default chat options 1`] = `
Object {
  "clientId": "CLIENT_ID",
  "onAuthenticationFailure": undefined,
  "token": "TOKEN",
}
`;
